tests/oidc/Earthfile

```diff
@@ -0,0 +1,39 @@
+VERSION 0.8
+PROJECT earthly-technologies/core
+
+FROM --pass-args ..+base
+
+IMPORT .. AS tests
+
+WORKDIR /test
+
+test:
+    BUILD +test-aws
+
+test-aws:
+    BUILD +test-aws-success
+    BUILD +test-aws-failure
+
+test-aws-base:
+    COPY test-aws.sh .
+    COPY aws.earth Earthfile
+    ENV EARTHLY_EXEC_CMD=/test/test-aws.sh
+
+test-aws-success:
+    FROM +test-aws-base
+    # happy path
+    ARG ROLE_ARN="arn:aws:iam::404851345508:role/ido-oidc-test"
```

Comment:
`ido-oidc-test` :raised_eyebrow:

I could look the other way (since it might be a pain to change it to something more descriptive like `oidc-integration-test` since it's not testing you)

Reply:
This probably wasn't obvious from my comment but these will change once I can set up the environment in production https://github.com/earthly/earthly/pull/4096/files#r1592957105
So once this PR is approved I will:
* merge my BE changes
* set up the prod env (which would not be using my name)
* Update this test (which would not be using my name)